R/MixedModelsCommon.R

```diff
@@ -792,1293 +793,4580 @@
+      y = levels(plotData$samp$parameter)[2]
+    ) + thm
+  graph
+
+}
+.rstanPlotAcor  <- function(plotData, lags = 30) {
+  clrs     <-
+    rep_len(rstan:::rstanvis_aes_ops("chain_colors"),
+            plotData$nchains)
+  thm      <- rstan:::rstanvis_theme()
+  dots     <-
+    rstan:::.add_aesthetics(list(), c("size", "color", "fill"))
+  ac_dat   <-
+    rstan:::.ac_plotData(dat = plotData$samp,
+                         lags = lags,
+                         partial = FALSE)
+
+  dots$position <- "dodge"
+  dots$stat     <- "summary"
+  dots$fun.y    <- "mean"
+  y_lab         <- gettext("Avg. autocorrelation")
+  ac_labs       <- ggplot2::labs(x = "Lag", y = y_lab)
+  y_scale       <-
+    ggplot2::scale_y_continuous(breaks = seq(0, 1, 0.25))
+  base          <-
+    ggplot2::ggplot(ac_dat, ggplot2::aes_string(x = "lag", y = "ac"))
+  graph         <-
+    base + do.call(ggplot2::geom_bar, dots) + y_scale + ac_labs + thm
+
+  graph
+}
+
+
+.mmCustomChecks <- list(
+  collinCheck = function(dataset){
+    cor_mat       <- cor(apply(dataset,2,as.numeric))
+    diag(cor_mat) <- 0
+    cor_mat[lower.tri(cor_mat)] <- 0
+    nearOne <- 1 - abs(cor_mat) < sqrt(.Machine$double.eps)
+    if(any(nearOne)){
+      var_ind   <- which(nearOne, arr.ind = TRUE)
+      varNames <- paste("'", .unv(rownames(cor_mat)[var_ind[,"row"]]),"' and '", .unv(colnames(cor_mat)[var_ind[,"col"]]),"'", sep = "", collapse = ", ")
+      return(gettextf("The following variables are a linear combination of each other, please, remove one of them from the analysis: %s", varNames))
+    }
+  }
+)
+.mmDependenciesLMM   <-
+  c(
+    "dependentVariable",
+    "fixedEffects",
+    "randomEffects",
+    "randomVariables",
+    "method",
+    "bootstrap_samples",
+    "test_intercept",
+    "type"
+  )
+.mmDependenciesGLMM  <- c(.mmDependenciesLMM,
+                          "dependentVariableAggregation",
+                          "family",
+                          "link")
+.mmDependenciesBLMM  <-
+  c(
+    "dependentVariable",
+    "fixedEffects",
+    "randomEffects",
+    "randomVariables",
+    "warmup",
+    "iteration",
+    "adapt_delta",
+    "max_treedepth",
+    "chains",
+    "seed",
+    "setSeed"
+  )
+.mmDependenciesBGLMM <- c(.mmDependenciesBLMM,
+                          "dependentVariableAggregation",
+                          "family",
+                          "link")
+# texts and messages
+.mmMessageInterpretability <-
+  gettext("The intercept corresponds to the (unweighted) grand mean; for each factor with k levels, k - 1 parameters are estimated. Consequently, the estimates cannot be directly mapped to factor levels.")
+.mmMessageSingularFit <-
+  gettext("Model fit is singular. Specified random effects parameters (random intercepts and random slopes) cannot be estimated from the available data. Carefully reduce the random effects structure, but this practice might inflate the reported p-value, and invalidates the analysis.")
```

Comment:
this is a fine approach, but I'd put these in a separate file.

Reply:
ok, done